packages/core/src/connectivity/scp-cf/proxy-util.ts

```diff
@@ -200,13 +200,13 @@ export function parseProxyEnv(
   proxyEnvValue: string
 ): ProxyConfiguration | undefined {
   const regex =
-    /(?<protocolWithDelimiter>(?<protocol>^.+):\/\/)?(?<userPwdWithDelimeter>(?<user>.+):(?<pwd>.+)@)?(?<hostAndPort>(?<host>[\w.]+):?(?<port>.+)?)/;
+    /(?<protocolWithDelimiter>(?<protocol>^\w+):\/\/)?(?<userPwdWithDelimiter>(?<user>.+):(?<pwd>.+)@)?(?<hostAndPort>(?<host>[\w.-]+):?(?<port>.+)?)/;
```

Comment:
The regex is too complicated so I can only rely on the tests.
I wonder, why don't we use the node native [URL class](https://nodejs.org/api/url.html#url_url). Maybe I miss something?
Here is the playground [tests](https://www.typescriptlang.org/play?ts=4.1.5#code/GYVwdgxgLglg9mABABwIYCcDOBTAqugGwAo11UBbALkUynRjAHMBKa3AJQBkBvAWAChEiCAlooMOACaIAvIjDYA7og6cSGCswDcAoSLCY4BbADoCcRuqzZJJ5OjhQ4Igtt3DRR0+cukpJgAs4WjdBDwMvMwsrf2Q4dChQvU9jKN8JGxMQHHQwCmwk8MNUnxjMtExMRXjJQvRsKBBc8WtJHX4AXwEBPzxCIgByAKgoZEoAenHsgFpsdEoKzGnq9EkAAUNybGmg2koADgAGI4HQ3vxiIZGxyahsPZOzjIvBu9oTEXJT9u7+IA), where shows you might have to handle the special case by our own:
- port number is 80, see [doc](https://nodejs.org/api/url.html#url_url_port)
- protocol is missing

I know these two special case also cost effort from our side, but getting rid of complicated regex might be better in my point of view.

Reply:
I was not aware, that the URL class also handles the user and password. I guess this should be fine. The only thing we would lose is the log statements in case the password and or user contain unencoded characters.

Reply:
besides that I would also say the library is better than the ugly regex

Reply:
Done.